sdk/core/azure-core/azure/core/rest/_helpers.py

```diff
@@ -285,22 +280,31 @@ def from_pipeline_transport_request_helper(request_class, pipeline_transport_req
     )
 
 def get_charset_encoding(response):
+    # type: (...) -> Optional[str]
     content_type = response.headers.get("Content-Type")
 
     if not content_type:
         return None
     _, params = cgi.parse_header(content_type)
     encoding = params.get('charset') # -> utf-8
-    if encoding is None:
-        if content_type in ("application/json", "application/rdap+json"):
-            # RFC 7159 states that the default encoding is UTF-8.
-            # RFC 7483 defines application/rdap+json
-            encoding = "utf-8"
-        else:
-            try:
-                encoding = chardet.detect(response.content)["encoding"]
-            except ResponseNotReadError:
-                pass
     if encoding is None or not lookup_encoding(encoding):
         return None
     return encoding
+
+def decode_to_text(encoding, content):
+    # type: (Optional[str], bytes) -> str
+    if encoding == "utf-8":
+        encoding = "utf-8-sig"
+    if encoding:
+        return content.decode(encoding)
+    # 1. We try utf-8-sig
+    decoder = codecs.getincrementaldecoder("utf-8-sig")(errors="strict")
```

Comment:
followed httpx's lead, and included trying to deserialize with windows 1252 @lmazuel . Lmk if we want to remove this

https://github.com/encode/httpx/blob/master/httpx/_models.py#L1317

Reply:
If utf-8-sig didn't work, utf-8 won't as well (so no need to do it later), utf-8-sig is a super-set of utf-8.

I wouldn't try cp1252 yet, httpx does it for HTML pages in old MS servers (we don't do that). Let's not fix what is not broken :)

Reply:
ok sounds good, will simplify my code too. thanks!